### ISSUE TITLE
I got an OOM error in a gan model.

### ISSUE THREAD

**Issue by paulbonnard** [keras-team/keras#12621]
Hello everyone,

I'm working on images. I made a gan model with keras to work on these images but i got an OOM error on the generative part.

I tried to reduce the batch (until 1) i reduced the images size too, i also reduced the numbers of layers of my model but i always got this error. I thought the function predict ignored the batch size, so i tried to put only 10 images in my files (size of one image = (400,500,3)) and also try the function predict_on_batch but same.

When i look at the summarize of the generative model i see 31k parameters, this is nothing.

i also launched my code on an aws instance with 8 gpu and 480 gigas of ram (lol) i got the error too ...

Do you have any idea why ?

Here is the (reduced) generator model :
<img width="398" alt="gen" src="https://user-images.githubusercontent.com/49201658/55570370-98b49280-5703-11e9-905d-e145a156b269.png">

And here the part of the train wich get an error : 

    `def train(epochs, batch_size, input_dir_low_res, input_dir_high_res):
       temps_debut = time.time()
       images_high_res = glob.glob(input_dir_high_res)
       images_low_res = glob.glob(input_dir_low_res)

       t = len(images_high_res)

      im_hr = plt.imread(images_high_res[0])
      n1,m1,c1 = np.shape(im_hr)
      image_shape_hr = (n1,m1,c1)

      im_lr = plt.imread(images_low_res[0])
      n2,m2,c2 = np.shape(im_lr)
      image_shape_lr = (n2,m2,c2)

      generator = Generator(image_shape_lr).generator() #build_generator(image_shape_lr, nombre_blocks 
      = 16, plot = 1)
   
      discriminator = Discriminator(image_shape_hr).discriminator()

  
      optimizer = get_optimizer()
      generator.compile(loss='mse', optimizer=optimizer)
      discriminator.compile(loss="binary_crossentropy", optimizer=optimizer)

      gan = get_gan_network(discriminator, image_shape_hr, generator, optimizer, 'mse')
  
      batch_count = int(t / batch_size)

      for e in range(1, epochs+1):
         print ('-'*15, 'Epoch %d' % e, '-'*15)
         for _ in tqdm(range(batch_count)):
           rand_nums = np.random.randint(0, t, size=batch_size)
           image_batch_hr = load_data(images_high_res,rand_nums)
           image_batch_hr = (image_batch_hr.astype(np.float32) - 127.5)/127.5 

           image_batch_lr = load_data(images_low_res,rand_nums)
           image_batch_lr = (image_batch_lr.astype(np.float32) - 127.5)/127.5
      
           generated_images_sr = generator.predict_on_batch(image_batch_lr)`

The error is on the last line

**Comment by nateraw** [comment]
Please fix formatting of code block. Make sure there are 3 backticks on each side of the code and that on the first one you have labelled it python. [Example here.](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code) Little hard to read/help otherwise. Thanks!

**Comment by gentlezr** [comment]
Howdy, maybe your batchsize is too big and try to reduce batchsize.
